seabird_service.proto

```diff
@@ -0,0 +1,109 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+import "common.proto";
+
+package seabird_chat_router;
+
+message JoinChannelEvent {
+  string channel_name = 1;
+}
+
+message LeaveChannelEvent {
+  string channel_id = 1;
+}
+
+message ChatEvent {
+  oneof inner {
+    common.MessageEvent text = 1;
+    common.PrivateMessageEvent private_message = 2;
+    common.MentionEvent mention = 3;
+    common.CommandEvent command = 4;
+
+    JoinChannelEvent join_channel = 5;
+    LeaveChannelEvent leave_channel = 6;
+  }
+}
+
+message RegisterRequest {
+  common.Identity identity = 1;
+
+  string name = 2;
+  repeated string supported_actions = 3;
+
+  // TODO: decide if this is worth the weirdness to
+  // save the extra RPC
+  repeated string initial_channels = 4;
+}
+
+message RegisterResponse {}
+
+message RegisterChannelsRequest {
+  common.Identity identity = 1;
+
+  repeated string channel_names = 2;
+}
+
+message RegisterChannelsResponse {
+  // Should the chat service maintain IDs as well?
```

Comment:
@belak looking for your input here.

Reply:
Is there a way for us to create a separate instance of (for example) the Discord plugin on the Seabird service for each explicitly defined use? The idea being that the upstream {Discord,IRC,Minecraft} service can already distinguish two different connections. If it can do that, do we need to deal with this at all? Basically I'm thinking that the Seabird service should be the only one that needs to care about this and downstream implementations of seabird-core need not even know this problem exists.

Reply:
At least in my opinion, the chat service should avoid IDs - it shouldn't even have to know about them.

Each chat provider would only be responsible for telling OP core what happened ("Joined a channel" or "incoming message"). Clients would be responsible for responding to messages and performing actions. Core would be the broker making that possible.

Reply:
Er, not implementations of seabird-core, but rather the plugins/middleware/adapters.